docs/proposals/11-scheduling.md

```diff
@@ -0,0 +1,133 @@
+
+`Reservation` resource: 
+```
+apiVersion: core.ironcore.dev/v1alpha1
+kind: Reservation
+metadata:
+  namespace: default
+  name: reservation
+spec:
+  # Pools define each pool where resources should be reserved
+  pools: 
+    - poolA
+    - poolB
+    - poolC
+  # Resources defines the requested resources in a ResourceList
+  resources:
+    localStorage: 10G
+    nics: 2
+    cpu: 2
+status:
+  # Pools describe where the resource reservation was possible
+  pools: 
+    - name: poolA
+      rating: 2
+      state: accepting
+    - name: poolB
+      ratring: 1
+      state: rejecting
+```
+
+Added `IRI` methods:
+
+```protobuf
+rpc ListReservations(ListReservationsRequest) returns (ListReservationsResponse) {};
+rpc CreateReservation(CreateReservationRequest) returns (CreateReservationResponse) {};
+rpc DeleteReservation(DeleteReservationRequest) returns (DeleteReservationResponse) {};
+```
+
+
+### Detailed flow
+1. If `.spec.machinePoolRef` is not set, the scheduler creates a `Reservation` that includes the required resources.
+2. `poollet`s which match `.spec.pools` of `Reservation` pick it up and broker it one layer down
+3. `poriver` evaluates the `Reservation` and sets its state to `accepting` or `rejecting`, which is then propagated up the hierarchy.
+4. The scheduler will pick a `pool` on every layer and update the `.spec.machinePoolRef` of the related `Machine`
+5. Once a `Machine` reaches a `provider` with a relating `Reservation`, the `Reservation` will be replaced through 
+   the `Machine`. Deletion process of `Reservation` is initiated through setting state to `releasing`, which is 
+   propagated up.
+6. Scheduler will delete top `Reservation` which triggers deletion of all brokered `Reservation`s 
+
+
+
+### Advantages
+- `Reservation`s can also be used to block resources for a specific use-case
+- In case of a partial outage/network issues `Machine`s can be placed on other `Pool`s
+- `Pool`s do not leak resource information, owner of resources decides if `Reservation` can be fulfilled (less complexity for over provisioning)
+
+### Disadvantages
+- One more resource (`Reservations`) is being introduced
```

Comment:
Increases complexity in the scheduling by introducing a new resource (`Reservation`) and the scheduler flow has to be extended.